### PR TITLE
Pubdev 5551 -  zero vs nas representation in sparse matrices

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-5551

### CODE REVIEW

**Comment by Pscheidl** [comment]
The last test added fails. Dense matrix distinguishes between zero and NaN and returns correct value. Sparse representation does not and unifies 0 with NaN. 

- Input data matrix is ok, the NaN is present there (fixed our code).
- Need to check indices and headers.

File: h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_zero_nan_handling_sparse_vs_native.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import xgboost as xgb
+
+from h2o.estimators.xgboost import *
+from tests import pyunit_utils
+
+
+def xgboost_categorical_zero_nan_handling_numerical_test():
```
Comment (Pscheidl) [inline h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_zero_nan_handling_sparse_vs_native.py]: For some reason, output of this test is:

```plain
[25. 40. 25.]
Parse progress: |█████████████████████████████████████████████████████████| 100%
Parse progress: |█████████████████████████████████████████████████████████| 100%
xgboost Model Build progress: |███████████████████████████████████████████| 100%
xgboost prediction progress: |████████████████████████████████████████████| 100%
  predict
---------
       30
       40
       30
```

Which is not right. Both native XGBoost and H2O treated `0` and `NaN` as the same value. But native XGBoost ends up in the middle between `20` and `30` (resulting in `25`), while H2O with learning rate 1 immediately learns the new value `30` instead of `20`.

**Comment by Pscheidl** [comment]
Unfortunately, this PR has been rebased onto master by mistake. I'll do a new one from latest rel-wolpert.
